share-coordinator/src/main/java/org/apache/kafka/coordinator/share/SharePartitionOffsetManager.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.coordinator.share;
+
+import org.apache.kafka.server.share.SharePartitionKey;
+import org.apache.kafka.timeline.SnapshotRegistry;
+import org.apache.kafka.timeline.TimelineHashMap;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class SharePartitionOffsetManager {
+
+    // map to store share partition key => current partition offset
+    // being written
+    private final TimelineHashMap<SharePartitionKey, Long> offsets;
+
+    // minimum offset representing the smallest necessary offset (non-redundant)
+    // across the internal partition
+    private long minOffset = Long.MAX_VALUE;
```

Comment:
I'm surprised that 0 is not a better starting value.

Reply:
This variable is used as arg in call to `Math.min` - 0 might not be suitable for that.

Reply:
Of course. Long.MAX_VALUE would be a catastrophic offset to delete from 😨